Platformdirs is not available on Solaris, and considering the thing it does is
very simple, it seems unnecessary to integrate it at this time.

--- pylint-2.15.0/pylint/constants.py
+++ pylint-2.15.0/pylint/constants.py
@@ -11,7 +11,6 @@ import sys
 from datetime import datetime
 
 import astroid
-import platformdirs
 
 from pylint.__pkginfo__ import __version__
 from pylint.typing import MessageTypesFullName
@@ -51,7 +50,10 @@ MAIN_CHECKER_NAME = "main"
 USER_HOME = os.path.expanduser("~")
 # TODO: 3.0: Remove in 3.0 with all the surrounding code
 OLD_DEFAULT_PYLINT_HOME = ".pylint.d"
-DEFAULT_PYLINT_HOME = platformdirs.user_cache_dir("pylint")
+path = os.environ.get("XDG_CACHE_HOME", "")
+if not path.strip():
+    path = os.path.expanduser("~/.cache")
+DEFAULT_PYLINT_HOME = os.path.join(path, "pylint")
 
 DEFAULT_IGNORE_LIST = ("CVS",)
 
--- pylint-2.15.0/pyproject.toml
+++ pylint-2.15.0/pyproject.toml
@@ -33,7 +33,6 @@ classifiers = [
 requires-python = ">=3.7.2"
 dependencies    = [
     "dill>=0.2",
-    "platformdirs>=2.2.0",
     # Also upgrade requirements_test_min.txt if you are bumping astroid.
     # Pinned to dev of second minor update to allow editable installs and fix primer issues,
     # see https://github.com/PyCQA/astroid/issues/1341
--- pylint-2.15.0/tests/lint/unittest_lint.py
+++ pylint-2.15.0/tests/lint/unittest_lint.py
@@ -21,7 +21,6 @@ from os.path import abspath, dirname, jo
 from pathlib import Path
 from shutil import rmtree
 
-import platformdirs
 import pytest
 from pytest import CaptureFixture
 
@@ -643,6 +642,7 @@ def pop_pylintrc() -> None:
     os.environ.pop("PYLINTRC", None)
 
 
+@pytest.mark.skip("platformdirs is not available on Solaris")
 @pytest.mark.usefixtures("pop_pylintrc")
 def test_pylint_home() -> None:
     uhome = os.path.expanduser("~")
